vm/actor/tests/paych_actor_test.rs

```diff
@@ -0,0 +1,1192 @@
+        rt.expect_send(
+            state.from,
+            METHOD_SEND,
+            Serialized::default(),
+            sent_to_from,
+            Serialized::default(),
+            exp_send_from_vec[i],
+        );
+        rt.expect_send(
+            state.to,
+            METHOD_SEND,
+            Serialized::default(),
+            state.to_send,
+            Serialized::default(),
+            exp_send_to_vec[i],
+        );
+
+        // Collect.
+        rt.set_caller(ACCOUNT_ACTOR_CODE_ID.clone(), state.from);
+        rt.expect_validate_caller_addr(&[state.from, state.to]);
+        let error = rt
+            .call(
+                &PAYCH_ACTOR_CODE_ID.clone(),
+                Method::Collect as u64,
+                &Serialized::default(),
+            )
+            .unwrap_err();
+        assert_eq!(exp_collect_vec[i], error.exit_code());
+    }
+}
+
+fn require_create_cannel_with_lanes<'a, BS: BlockStore>(
+    bs: &'a BS,
+    num_lanes: u64,
+) -> (MockRuntime<'a, BS>, SignedVoucher) {
+    let paych_addr = Address::new_id(100);
+    let payer_addr = Address::new_id(102);
+    let payee_addr = Address::new_id(103);
+    let balance = TokenAmount::from(100_000 as u64);
+    let recieved = TokenAmount::from(0 as u64);
+
+    let curr_epoch = 2;
+
+    let message = UnsignedMessage::builder()
+        .from(*SYSTEM_ACTOR_ADDR)
+        .to(paych_addr)
+        .gas_limit(1000)
+        .build()
+        .unwrap();
+
+    let mut rt = MockRuntime::new(bs, message);
+
+    rt.epoch = curr_epoch;
+    rt.balance = TokenAmount::from(balance);
+    rt.received = TokenAmount::from(recieved);
+    rt.set_caller(INIT_ACTOR_CODE_ID.clone(), *INIT_ACTOR_ADDR);
+    rt.actor_code_cids
+        .insert(payee_addr, ACCOUNT_ACTOR_CODE_ID.clone());
+    rt.actor_code_cids
+        .insert(payer_addr, ACCOUNT_ACTOR_CODE_ID.clone());
+    construct_and_verify(&mut rt, payer_addr, payee_addr);
+
+    let mut last_sv = SignedVoucher::default();
+    for i in 0..num_lanes {
+        let lane_param = LaneParams {
+            epoch_num: curr_epoch,
+            from: payer_addr,
+            to: payee_addr,
+            amt: (i + 1) as i64,
+            lane: i as u64,
+            nonce: i + 1,
+        };
+
+        last_sv = require_add_new_lane(&mut rt, lane_param);
+    }
+
+    (rt, last_sv)
+}
+
+fn require_add_new_lane<BS: BlockStore>(
+    rt: &mut MockRuntime<'_, BS>,
+    param: LaneParams,
+) -> SignedVoucher {
+    let payee_addr = Address::new_id(103);
+    let sig = Signature::new_bls(vec![0, 1, 2, 3, 4, 5, 6, 7]);
+    let sv = SignedVoucher {
+        time_lock_min: param.epoch_num,
+        time_lock_max: u64::MAX,
+        secret_pre_image: vec![],
+        extra: Option::None,
+        lane: param.lane,
+        nonce: param.nonce,
+        amount: BigInt::from(param.amt),
+        min_settle_height: 0,
+        merges: vec![],
+        signature: Some(sig.clone()),
+    };
```

Comment:
SignedVoucher can probably have a default implementation derived, to avoid having to construct all of this when you are just setting a few values, and just defaulting all ignored fields with `..Default::default()`

Reply:
ye should shave off a few lines